kde-misc/kdiff3/patches/kdiff3-0.9.98.patchset

```diff
@@ -0,0 +1,41 @@
+From 0fe9538680880480f833fae95b96be7747714ac0 Mon Sep 17 00:00:00 2001
+From: Bach Nguyen <bach5000@gci.test>
+Date: Tue, 26 Dec 2017 17:44:53 +0000
+Subject: Fix settings dir
+
+
+diff --git a/src-QT4/kreplacements/kreplacements.cpp b/src-QT4/kreplacements/kreplacements.cpp
+index 6564ae8..0fd2261 100644
+--- a/src-QT4/kreplacements/kreplacements.cpp
++++ b/src-QT4/kreplacements/kreplacements.cpp
+@@ -48,6 +48,9 @@
+ #include <vector>
+ #include <algorithm>
+ 
++#if defined(__HAIKU__)
++#include <FindDirectory.h>
++#endif
+ 
+ static QString s_copyright;
+ static QString s_email;
+@@ -422,9 +425,16 @@ QString KStandardDirs::findResource(const QString& resource, const QString& /*ap
+       QString configPath = exePath + "/.kdiff3rc"; // This is useful for portable installations (e.g. on USB-Stick)
+       if ( QFile::exists( configPath ) )
+          return configPath;
+-#endif
++#elif defined(__HAIKU__)
+       QString home = QDir::homePath();
++      char hdir[B_PATH_NAME_LENGTH + B_FILE_NAME_LENGTH];
++      find_directory(B_USER_SETTINGS_DIRECTORY, 0, false, hdir, 512);
```

Comment:
Please also change the other 512

Also, as noted, the constant for our `/boot` is [`-1`](http://xref.plausible.coop/source/xref/haiku/src/system/libroot/os/find_directory.cpp#277), not `0`

To avoid fiddling with constants, I suggest using the [C++ interface](https://www.haiku-os.org/docs/api/FindDirectory_8h.html#a87ac32df2ed40b7991ade10a2fa375b1)

```diff
-++      find_directory(B_USER_SETTINGS_DIRECTORY, 0, false, hdir, 512);
+++      find_directory(B_USER_SETTINGS_DIRECTORY, -1, false, hdir, sizeof(hdir));
```